lib/internal/child_process.js

```diff
@@ -456,16 +456,22 @@ function setupChannel(target, channel) {
       }
       chunks[0] = jsonBuffer + chunks[0];
 
+      var nextTick = false;
       for (var i = 0; i < numCompleteChunks; i++) {
         var message = JSON.parse(chunks[i]);
```

Comment:
I meant implement a string based check. Can't we assume "innerMessages" will have a `{cmd: NODE_` prefix? or at least include a `, cmd: NODE_` substring? Might also be able to distinguish between three cases: user/inner/inner_with_handle, and turn the following compound `if` into a `switch`